synapse/handlers/register.py

```diff
@@ -609,15 +609,27 @@ async def _auto_join_rooms(self, user_id: str) -> None:
         if (
             self.hs.config.registration.autocreate_auto_join_rooms
             and await self.store.is_real_user(user_id)
+            and not user_id.startswith(
+                "bot_"
+            )  # Check if the user is a bot based on your specific logic
```

Comment:
This does not characterise bot users: users are free to register accounts like `@bot_mc_bot_face:example.org`.

I think the concept you're looking for here is an "appservice" user.

Reply:
> I think the concept you're looking for here is an "appservice" user.

Just quickly and randomly chiming in: it looks to me like the bot/support user distinction mentioned in #15813 is about [`UserType`s](https://github.com/matrix-org/synapse/blob/52d8131e87f90114018cac6daf01b1dd64eac28b/synapse/api/constants.py#L171) rather than appservice users. Though I haven't look further than this diff and the issue description, and maybe the same issue applies to AS users.